mlrun/api/api/endpoints/functions.py

```diff
@@ -706,13 +706,14 @@ def _build_function(
             reason=f"runtime error: {err_to_str(err)}",
         )
     try:
+        is_nuclio_runtime = bool(fn.kind in RuntimeKinds.nuclio_runtimes())
 
-        # enrich runtime with project defaults
+        # Enrich runtime with project defaults
         launcher = mlrun.api.launcher.ServerSideLauncher(auth_info=auth_info)
-        launcher.enrich_runtime(runtime=fn)
+        launcher.enrich_runtime(runtime=fn, full=is_nuclio_runtime)
```

Comment:
Maybe add a comment explaining what this means?
If the runtime is nuclei then you need to enrich the project's full spec because...?